client/rpcproxy/server_endpoint.go

```diff
@@ -0,0 +1,81 @@
+package rpcproxy
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+const (
+	defaultNomadRPCPort = "4647"
+)
+
+// EndpointKey is used in maps and for equality tests.  A key is based on endpoints.
+type EndpointKey struct {
+	name string
+}
+
+// Equal compares two EndpointKey objects
+func (k *EndpointKey) Equal(x *EndpointKey) bool {
+	return k.name == x.name
+}
+
+// ServerEndpoint contains the address information for to connect to a Nomad
+// server.
+//
+// TODO(sean@): Server is stubbed out so that in the future it can hold a
+// reference to Node (and ultimately Node.ID).
+type ServerEndpoint struct {
+	// Name is the unique lookup key for a Server instance
+	Name string
+	Host string
+	Port string
+	Addr net.Addr
+}
+
+// Key returns the corresponding Key
+func (s *ServerEndpoint) Key() *EndpointKey {
+	return &EndpointKey{
+		name: s.Name,
+	}
+}
+
+// newServer creates a new Server instance with a resolvable endpoint
+func newServer(name string) (s *ServerEndpoint, err error) {
```

Comment:
Add documentation such that name can be either an address or dns name  and remove the variable names in the return